tidb-cloud/data-service-oas-with-nextjs.md

```diff
@@ -0,0 +1,205 @@
+---
+title: Use the OpenAPI Specification of a Data App with Next.js
+summary: Learn how to use the OpenAPI Specification of a Data App to generate client code and develop a Next.js application.
+---
+
+# Use the OpenAPI Specification of a Data App with Next.js
+
+## Before you begin
+
+Before using OpenAPI Specification with Next.js, make sure that you have the following:
+
+- A [TiDB Serverless cluster](/tidb-cloud/create-tidb-cluster-serverless.md)
+<!--TODO: add yarn-->
```

Comment:
- Install node, [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) and [yarn](https://yarnpkg.com/getting-started/install)